### ISSUE TITLE
Error#build_error_message returns hard-to-parse string.

### ISSUE THREAD

**Issue by timobleeker** [PromisePay/promisepay-ruby#25]
The errors the gem returns aren't easily parsable. 
E.g.: `"name: [\"can't be blank\", \"required field missing\"]"`

A hash would be much more valuable IMHO. It's easier to iterate over, and it's a common structure in Rails.
This would be a matter of removing the following bit of code in the `Error#build_error_message` method:

``` ruby
    def build_error_message
      return nil if @response.nil? || @response.body.nil?

      json_response = JSON.parse(@response.body)
# Remove from here...
      message = ''
      message << json_response['message'] if json_response.key?('message')
      if json_response.key?('errors')
        json_response['errors'].each do |attribute, content|
          message << "#{attribute}: #{content}"
        end
      end
      message
# ...to here.
    end
  end
```

**Comment by romsssss** [comment]
Hi Tim,

Thanks for raising this issue !
The original intent of formatting the errors message this way was to be as close as possible to what `ActiveRecord` does.
Now, I agree with you, this is not the easiest format to parse.

I just added (https://github.com/PromisePay/promisepay-ruby/pull/28) a new option for the client called `:errors_format`. This options has two values:
* *processed*: errors messages will be formatted as they currently are
* *raw*: errors messages won't be formatted at all and could be easily parsed.

For backward compatibility purpose, default value is *processed*.

Like the other ones, this option can be either set with a global variable
````
export PROMISEPAY_ERRORS_FORMAT=raw
````
or whenever instantiating the client
````ruby
client = Promisepay::Client.new(errors_format: 'raw')
````

Using *raw* errors formatting, the following is now possible:
````ruby
client = Promisepay::Client.new(errors_format: 'raw')

begin
  client.users.create(
    id: '123456',
    # first_name: 'test',
    # last_name: 'buyer',
    email: 'buyer@test.com',
    address_line1: '48 collingwood',
    state: 'vic',
    city: 'Mel',
    zip: '3000',
    country: 'AUS',
    dob:'12/06/1980'
  )
rescue Promisepay::UnprocessableEntity => e
  errors = JSON.parse(e.message)
  errors['errors']['name']
end
=> ["can't be blank", "required field missing"]
```

Please note that this feature will be available in the next release of the gem (in the coming days).

**Comment by timobleeker** [comment]
Hey @romsssss, 
thanks, I think this will make it a lot easier for me to handle the errors. 

> The original intent of formatting the errors message this way was to be as close as possible to what ActiveRecord does.
As far as I know ActiveRecord returns a `Hash` containing model/validation errors, so I'm not sure if I follow?

**Comment by romsssss** [comment]
You are right, [ActiveModel::Errors](http://api.rubyonrails.org/classes/ActiveModel/Errors.html) is a modified Hash and the original errors implementation in this SDK was closer to  [ActiveModel::Errors#full_messages](http://api.rubyonrails.org/classes/ActiveModel/Errors.html#method-i-full_messages) method.

Anyway, we released [v1.1.0](https://rubygems.org/gems/promisepay/versions/1.1.0) of the gem last week which includes the new errors formatting: https://github.com/PromisePay/promisepay-ruby/pull/28

**Comment by ngelx** [comment]
Nice patch. This should be in the doc. It is very useful.
